src/main/java/com/joyent/manta/client/MantaHttpHeaders.java

```diff
@@ -0,0 +1,161 @@
+package com.joyent.manta.client;
+
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.util.FieldInfo;
+import com.google.api.client.util.Types;
+import org.apache.http.Header;
+import org.apache.http.HeaderElement;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.message.BufferedHeader;
+import org.apache.http.protocol.HTTP;
+
+import java.util.*;
+
+import static com.joyent.http.signature.HttpSignerUtils.X_REQUEST_ID_HEADER;
+
+/**
+ * Object encapsulating the HTTP headers to be sent to the Manta API.
+ * When non-standard HTTP headers are used as part of a PUT request to
+ * Manta, they are stored as metadata about an object.
+ *
+ * @author <a href="https://github.com/dekobon">Elijah Zupancic</a>
+ */
+public class MantaHttpHeaders extends HttpHeaders {
```

Comment:
The problem with having this inherit from HttpHeaders is that HttpHeaders contains a lot of methods which return itself so it can be manipulated fluently.  These methods return `com.google.api.client.http.HttpHeaders` instead of `com.joyent.manta.client. MantaHttpHeaders` which messes up chained invocation :disappointed: